include/aspect/volume_of_fluid/utilities.h

```diff
@@ -0,0 +1,152 @@
+
+ You should have received a copy of the GNU General Public License
+ along with ASPECT; see the file doc/COPYING.  If not see
+ <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef _aspect__volume_of_fluid_utilities_h
+#define _aspect__volume_of_fluid_utilities_h
+
+#include <deal.II/base/point.h>
+
+#include <aspect/global.h>
+
+namespace aspect
+{
+  namespace VolumeOfFluid
+  {
+    using namespace dealii;
+
+    // Utilities
+
+    /**
+     * Function to calculate volume fraction contained by indicator function
+     * H(d-normal*(x'-x_{cen}')) on the [0, 1]^dim unit cell where x_{cen} is
+     * the unit cell center.
+     *
+     * Currently only works assuming constant Jacobian determinant.
+     */
+    template<int dim>
+    double compute_fluid_fraction (const Tensor<1, dim, double> normal,
+                                   const double d);
+
+    /**
+     * Function to calculate required value of d to obtain given volume
+     * fraction for indicator function H(d-normal*(x'-x_{cen}')) on the [0,
+     * 1]^dim unit cell where x_{cen} is the unit cell center.
+     *
+     * Currently only works assuming constant Jacobian determinant.
+     */
+    template<int dim>
+    double compute_interface_location (const Tensor<1, dim, double> normal,
+                                       const double volume_fraction);
+
+    /**
+     * Obtain values at points for a polynomial function that is equivalent to
+     * the Heaviside function $H(d-normal*xhat)$ on the unit cell when
+     * integrated against polynomials of up to the specified degree.
+     *
+     * Currently works for degree <=1
+     *
+     * @param degree Maximum degree for exact integration
+     * @param normal Interface normal vector, pointed away from the included region
+     * @param d Interface parameter specifying location of interface in unit cell
+     * @param points Locations to evaluate the constructed polynomial
+     * @param values Values of the constructed polynomial at the specified points
+     */
+    template<int dim>
+    void xFEM_Heaviside(const int degree,
+                        const Tensor<1, dim, double> normal,
+                        const double d,
+                        const std::vector<Point<dim>> &points,
+                        std::vector<double> &values);
+    /**
+     * Obtain values at points for a polynomial function that is equivalent to
+     * the function $\frac{d}{dd}H(d-normal*xhat)$ on the unit cell when
+     * integrated against polynomials of up to the specified degree.
+     *
+     * Currently works for degree <=1
+     *
+     * @param degree Maximum degree for exact integration
+     * @param normal Interface normal vector, pointed away from the included region
+     * @param d Interface parameter specifying location of interface in unit cell
+     * @param points Locations to evaluate the constructed polynomial
+     * @param values Values of the constructed polynomial at the specified points
```

Comment:
This documentation is identical to the one above, please clarify what is different about the two functions.

Reply:
Re-read the documentation. One is for the equivalent to the heaviside, the other is the derivative.